dango/account/margin/src/core.rs

```diff
@@ -0,0 +1,183 @@
+        // is zero.
+        if collateral_balance.is_zero() {
+            continue;
+        }
+
+        let price = querier.query_price(app_cfg.addresses.oracle, denom, None)?;
+        let value = price.value_of_unit_amount(collateral_balance)?;
+        let adjusted_value = value.checked_mul(power.clone().into_inner())?;
+
+        collaterals.insert(Coin::new(denom.clone(), collateral_balance)?)?;
+        total_collateral_value.checked_add_assign(value)?;
+        total_adjusted_collateral_value.checked_add_assign(adjusted_value)?;
+    }
+
+    // ---------------------------- 3. Limit Orders ----------------------------
+
+    // Add assets locked in limit orders to the total adjusted collateral value.
+    //
+    // For BUY orders, the user have transferred the quote asset to the DEX;
+    // conversely, for SELL orders, the user have transferred the base asset.
+    //
+    // The collateral value of a limit order is evaluated as either that of the
+    // input asset, or that of the output asset, whichever is smaller.
+    let mut limit_order_collaterals = Coins::new();
+    let mut limit_order_outputs = Coins::new();
+
+    // Query the user's open limit orders.
+    let orders = querier.query_wasm_smart(app_cfg.addresses.dex, QueryOrdersByUserRequest {
+        user: account,
+        start_after: None,
+        limit: None,
+    })?;
+
+    // Iterate over the user's limit orders and add the order value to the
+    // total collateral value.
+    for (_, res) in orders {
+        // Get asset locked in the order and the asset that would be returned
+        // if the order was filled.
+        let (deposited, returned) = match res.direction {
+            Direction::Bid => (
+                Coin::new(
+                    res.quote_denom.clone(),
+                    res.remaining.checked_mul_dec_ceil(res.price)?,
+                )?,
+                Coin::new(res.base_denom.clone(), res.remaining)?,
+            ),
+            Direction::Ask => (
+                Coin::new(res.base_denom.clone(), res.remaining)?,
+                Coin::new(
+                    res.quote_denom.clone(),
+                    res.remaining.checked_mul_dec_floor(res.price)?,
+                )?,
+            ),
+        };
+
+        let deposited_price =
+            querier.query_price(app_cfg.addresses.oracle, &deposited.denom, None)?;
+        let returned_price =
+            querier.query_price(app_cfg.addresses.oracle, &returned.denom, None)?;
+        let deposited_value = deposited_price.value_of_unit_amount(deposited.amount)?;
+        let returned_value = returned_price.value_of_unit_amount(returned.amount)?;
+
+        let min_value = min(deposited_value, returned_value);
+
+        let collateral_power = app_cfg
+            .collateral_powers
+            .get(&deposited.denom)
+            .ok_or_else(|| anyhow!("collateral power for denom {} not found", deposited.denom))?;
```

Comment:
Collateral power should just be zero if not found. Imagine someone launches a random shitcoin. Of course it won't have a collateral power. But people may still trade it on the DEX.

Reply:
True. Will fix

Reply:
fixed